deploy/docker/docker-entrypoint.sh

```diff
@@ -128,9 +128,9 @@ try_fill_config() {
     if grep -qE "^[#[:space:]]*$escaped_key\s*=" "$file"; then
         echo_value "$key" "$value"
         if [[ -z "$value" ]]; then
-            echo "$(sed -r "s/^[#[:space:]]*($escaped_key)\s*=\s*(.*)/# \1 = \2/" "$file")" > "$file"
+            sed -r "s/^[#[:space:]]*($escaped_key)\s*=\s*(.*)/# \1 = \2/" "$file" > tmpfile && mv tmpfile "$file"
```

Comment:
please verify if `mv` will work when config is in a mount volume in a docker container.
this was changed from `sed -i` into `echo ... >` due to the fact that `sed -i` creates a tmp file and `mv` to rename.

Reply:
It works. I verified it with the command
```
docker run --volume=/Users/yudai/tmp/etc/plugins:/opt/emqx/etc/plugins <image id>
```

Reply:
I never really understood what was the original issue which triggered the change from `sed -i`
now I found this: https://apimirror.com/docker~1.10/engine/userguide/containers/dockervolumes/index

> Note: Many tools used to edit files including vi and sed --in-place may result in an inode change. Since Docker v1.1.0, this will produce an error such as “sed: cannot rename ./sedKdJ9Dy: Device or resource busy”. In the case where you want to edit the mounted file, it is often easiest to instead mount the parent directory.

I am however still not sure why your approach is different from `sed -i`.
maybe we should `cat tmpfile > "$file"` ?
so to keep the inode and file owner.

Reply:
ok. I think I finally got it. it's when mount a file, not the parent directory.

Reply:
fixed to `sed -r "s/^[#[:space:]]*($escaped_key)\s*=\s*(.*)/# \1 = \2/" "$file" > tmpfile && cat tmpfile > "$file"`